### PR TITLE
check max validate depth

### PR DESCRIPTION
**Pull request by nevermore3**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

#### Issue(s) number: 
close https://github.com/vesoft-inc/nebula-ent/issues/3403
## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/graph/validator/Validator.h [hunk 1]
```diff
@@ -203,6 +204,7 @@ class Validator {
   std::set<std::string> userDefinedVarNameList_;
   // vid's Type
   Value::Type vidType_;
+  static thread_local uint32_t maxStatements_;
```
Comment (jievince) [inline src/graph/validator/Validator.h:207]: Why it need to be thread_local? Validator is single-thread.

Comment (nevermore3) [reply src/graph/validator/Validator.h:207]: makeValidator is a static function，maxStatement_ is also static member, so use thread_local

**Review by jievince — DISMISSED** [review]

File: src/graph/validator/Validator.cpp [hunk 2]
```diff
@@ -273,22 +281,27 @@ std::unique_ptr<Validator> Validator::makeValidator(Sentence* sentence, QueryCon
...
-  if (!root) {
-    return Status::SemanticError("Get null plan from sequential validator");
+    auto root = validator->root();
+    if (!root) {
+      return Status::SemanticError("Get null plan from sequential validator");
+    }
+    qctx->plan()->setRoot(root);
+    // reset maxStatements
+    maxStatements_ = 0;
+  } catch (const std::runtime_error& error) {
```
Comment (yixinglu) [inline src/graph/validator/Validator.cpp:301]: catch std::exception

Comment (nevermore3) [reply src/graph/validator/Validator.cpp]: done

**Review by yixinglu — APPROVED** [review]

**Review by jievince — APPROVED** [review]

**Pull request merged**